client/controllers/contacts.spec.js

```diff
@@ -0,0 +1,45 @@
+describe("The client-side ContactController", function() {
+    // module() is provided by angular-mocks.js
+    beforeEach(module("app"));
+    
+    var $controller, $httpBackend, $rootScope;
+    var contactsController, expectedAllContacts, expectedNewContact;
+    
+    // $controller is the service responsible for instantiating controllers.
+    // It is not available on the global scope, so we must inject it this way.
+    beforeEach(inject(function($injector) {
+        $httpBackend = $injector.get("$httpBackend");
+        $rootScope = $injector.get("$rootScope");
+        $controller = $injector.get("$controller");
+        
+        expectedAllContacts = readJSON("development-data.json").allContacts;
+        expectedNewContact = readJSON("development-data.json").newContact;
+        
+        $httpBackend.expectGET("/contacts").respond(200, expectedAllContacts);
+        contactsController = $controller("ContactController", {"$scope": $rootScope});
+        $httpBackend.flush();
+    }));
+    
+    afterEach(function() {
+        $httpBackend.verifyNoOutstandingExpectation();
+        $httpBackend.verifyNoOutstandingRequest();
+    });
+    
+    it("can get all contacts", function() {
+        $httpBackend.expectGET("/contacts").respond(200, expectedAllContacts);
+        contactsController.getContacts();
+        $httpBackend.flush();
+    });
+    
+    it("can add a new contact and update the list", function() {
+        contactsController.newContact = JSON.parse(JSON.stringify((expectedNewContact)));
```

Comment:
This `JSON.parse(JSON.stringify(expectedNewContact))` was necessary to copy the object. There may be better ways to do this in the future.

Also, remove the extra parentheses around `expectedNewContact`.